pkgs/checks/test/test_shared.dart

```diff
@@ -51,16 +57,17 @@ extension RejectionChecks<T> on Subject<T> {
       didRunCallback = true;
       final failure = await softCheckAsync(value, condition);
       if (failure == null) {
-        return Extracted.rejection(which: [
-          'was accepted by the condition checking:',
-          ...await describeAsync(condition)
-        ]);
+        final description = await describeAsync(condition);
```

Comment:
Here is one place where we are not getting value from the callback potentially not getting invoked.

I can't make the `which` callback async, so we are forced to do this async work outside of the callback.